staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -306,17 +306,24 @@ func TestApfExecuteMultipleRequests(t *testing.T) {
 	defer cancel()
 	StartPriorityAndFairnessWatermarkMaintenance(ctx.Done())
 
-	var wg sync.WaitGroup
-	wg.Add(concurrentRequests)
+	errGotCh := make(chan error, concurrentRequests)
+	defer close(errGotCh)
 	for i := 0; i < concurrentRequests; i++ {
 		go func() {
-			defer wg.Done()
-			if err := expectHTTPGet(fmt.Sprintf("%s/api/v1/namespaces/default", server.URL), http.StatusOK); err != nil {
-				t.Error(err)
-			}
+			errGotCh <- expectHTTPGet(fmt.Sprintf("%s/api/v1/namespaces/default", server.URL), http.StatusOK)
 		}()
 	}
-	wg.Wait()
+
+	for i := 0; i < concurrentRequests; i++ {
+		select {
+		case errGot := <-errGotCh:
+			if errGot != nil {
+				t.Errorf("Expected no error from concurrent request, but got: %v", errGot)
+			}
+		case <-time.After(wait.ForeverTestTimeout):
```

Comment:
I recommend calling this once and storing it and using it in every iteration rather than starting a new timeout every time through.

Reply:
I reverted changes to these tests